File: QID-3482-SFSbb/SFSbb.R
# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

# Brownian Bridge main calculation
dt = 0.004
n  = 250
l  = 1
t  = seq(0, n * dt, dt)

set.seed(0)
z      = runif(n, min = 0, max = 1)
z      = 2 * (z > 0.5) - 1
z      = z * sqrt(dt)  # to get finite and non-zero varinace
z2     = dt * sum(z)
z      = z - z2
x      = c(0, cumsum(z))
listik = cbind(t, x)

# Output
plot(listik[, 1], listik[, 2], type = "l", col = "blue", xlab = "Time", ylab = "Values of process X_t") 

File: QID-3482-SFSbb/SFSbb.m
%------------------------------------------------------------------------
% Book:         SFS
% ----------------------------------------------------------------------
% Quantlet:     SFSbb
% ----------------------------------------------------------------------
% Description:  A sample path of Brownian Bridge
%------------------------------------------------------------------------
% Usage:        SFSbb
%-----------------------------------------------------------------------
% Inputs:       none
%-----------------------------------------------------------------------
% Output:       a plot of a sample path
%               
% ----------------------------------------------------------------------
% Example:      the example of a sample path
%------------------------------------------------------------------------
% Author  :     Weining Wang
%------------------------------------------------------------------------

clear all;
close all;
clc;

%%Brownian Bridge
% main calculation
dt = 0.004;
n  = 250;
l  = 1;
t  = 0:dt:n*dt; 

%randseed(0);
z      = unifrnd(0,1,n,1);
z      = 2*(z>0.5)-1;
z      = z*sqrt(dt);  %//to get finite and non-zero varinace
z2     = dt*sum(z);
z      = z-z2;
x      = [0;cumsum(z)];
listik = [t',x];

% output
plot(listik(:,1),listik(:,2))

%title('Wiener process')
xlabel('Time')
ylabel('Values of process X_t')